ledger/src/blockstore_processor.rs

```diff
@@ -1673,14 +1676,30 @@ fn confirm_slot_entries(
         .expect("Transaction verification generates entries");
 
     let mut replay_timer = Measure::start("replay_elapsed");
+    let is_vote_only_bank = bank.vote_only_bank();
     let replay_entries: Vec<_> = entries
         .into_iter()
         .zip(entry_tx_starting_indexes)
-        .map(|(entry, tx_starting_index)| ReplayEntry {
-            entry,
-            starting_index: tx_starting_index,
+        .map(|(entry, tx_starting_index)| {
+            // If bank is in vote-only mode, validate that entries contain only vote transactions
+            if let EntryType::Transactions(ref transactions) = entry {
+                if is_vote_only_bank
+                    && transactions
+                        .iter()
+                        .any(|tx| !tx.is_simple_vote_transaction())
```

Comment:
thinking about performance, this seems fine. `is_vote_only_bank` always false except during the transition (in which case load is already lighter. And even then, I believe these are `RuntimeTransaction` types, so the is_simple_vote_transaction is super cheap

Reply:
Also, is the check on number of transactions is bounded at this point based on shred count and compute units?